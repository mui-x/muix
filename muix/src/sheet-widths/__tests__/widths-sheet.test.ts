import { TVariants } from "reactxx-typings";
import { atomizeSheet } from "reactxx-sheeter";
import { initPlatform, Shape, ts } from "reactxx-tests";
import { sheetWidths_registerVariantHandler } from "../index";

sheetWidths_registerVariantHandler();

export const createSheet = () =>
  (ts.sheet = {
    root: (ts.view = {
      $widths: {
        ['0-640']: {
          margin: 0,
          $switch: {
            isClosed: {
              backgroundColor: "gray"
            }
          }
        },
        ['640-1024']: {
          margin: 10
        },
        ['1024-']: {
          margin: 20
        }
      },
      $web: (ts.web = {
        ":hover": {
          $switch: {
            isClosed: {
              color: "black",
              $widths: {
                ['0-640']: {
                  color: "blue",
                  padding: 0
                },
                ['640-1024']: {
                  color: "red",
                  padding: 10
                },
                ['1024-']: {
                  padding: 20
                }
              }
            }
          }
        }
      })
    }),
    label: {},
    webOnly: {},
    nativeOnly: {}
  });
export const query = (opened: boolean) =>
  ({
    $widths: {
      tabletWidth: opened,
      mobileWidth: !opened
    }
  } as TVariants.Query<Shape>);
describe("WIDTHS define sheet", () => {
  it("WEB", () => {
    initPlatform(true);
    const sheet = atomizeSheet<Shape>(createSheet());
    expect(sheet).toMatchSnapshot();
  });
  it("NATIVE", () => {
    initPlatform(false);
    const sheet = atomizeSheet<Shape>(createSheet());
    expect(sheet).toMatchSnapshot();
  });
});
