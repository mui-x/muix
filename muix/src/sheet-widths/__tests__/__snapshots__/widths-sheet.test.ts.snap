// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WIDTHS define sheet NATIVE 1`] = `
Object {
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "margin: 0 /*sheet.root/$widths.0-640*/",
        ],
        "conditions": Array [
          Object {
            "interval": Array [
              0,
              640,
            ],
            "type": "$widths",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          "margin: 10 /*sheet.root/$widths.640-1024*/",
        ],
        "conditions": Array [
          Object {
            "interval": Array [
              640,
              1024,
            ],
            "type": "$widths",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          "margin: 20 /*sheet.root/$widths.1024-*/",
        ],
        "conditions": Array [
          Object {
            "interval": Array [
              1024,
              0,
            ],
            "type": "$widths",
          },
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`;

exports[`WIDTHS define sheet WEB 1`] = `
Object {
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "@media (max-width: 639px).a { margin:0 /*sheet.root/$widths.0-640*/ }",
        ],
      },
      Object {
        "atomicArray": Array [
          "@media (min-width: 640px) and (max-width: 1023px).b { margin:10px /*sheet.root/$widths.640-1024*/ }",
        ],
      },
      Object {
        "atomicArray": Array [
          "@media (min-width: 1024px).c { margin:20px /*sheet.root/$widths.1024-*/ }",
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`;
